sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -729,32 +729,55 @@ void gnrc_rpl_send_DAO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination, uint
         return;
     }
 
-    gnrc_ipv6_netif_addr_t *me_netif = gnrc_ipv6_netif_addr_get(me);
-    if (me_netif == NULL) {
-        DEBUG("RPL: no netif address found for %s\n",
-              ipv6_addr_to_str(addr_str, me, sizeof(addr_str)));
-        return;
+    fib_entry_t *fentry = NULL;
+    ipv6_addr_t *addr = NULL;
+
+    mutex_lock(&(gnrc_ipv6_fib_table.mtx_access));
+
+    /* add external and RPL FIB entries */
+    for (size_t i = 0; i < gnrc_ipv6_fib_table.size; ++i) {
+        fentry = &gnrc_ipv6_fib_table.data.entries[i];
+        if (fentry->lifetime != 0) {
+            if (!(fentry->next_hop_flags & FIB_FLAG_RPL_ROUTE)) {
+                ptr = &tmp;
+                if (!ext_processed) {
+                    if ((tmp = _dao_transit_build(NULL, lifetime, true)) == NULL) {
+                        DEBUG("RPL: Send DAO - no space left in packet buffer\n");
+                        mutex_unlock(&(gnrc_ipv6_fib_table.mtx_access));
+                        return;
+                    }
+                    ext_processed = true;
+                }
+            }
+            else {
+                ptr = &pkt;
+                if (!int_processed) {
+                    if ((tr_int = pkt = _dao_transit_build(NULL, lifetime, false)) == NULL) {
+                        DEBUG("RPL: Send DAO - no space left in packet buffer\n");
+                        mutex_unlock(&(gnrc_ipv6_fib_table.mtx_access));
+                        return;
+                    }
+                    int_processed = true;
+                }
+            }
+            addr = (ipv6_addr_t *) fentry->global->address;
```

Comment:
a closer look shows me that you don't change anything in the entries :sweat_smile: (phew)